app/controllers/miq_capacity_controller.rb

```diff
@@ -154,43 +154,28 @@ def change_tab
   end
 
   def planning_option_changed
-    if params[:filter_typ]
-      @sb[:planning][:vms] = nil
-      @sb[:planning][:options][:chosen_vm] = nil
-      @sb[:planning][:options][:filter_typ] = params[:filter_typ] == "<Choose>" ? nil : params[:filter_typ]
-      @sb[:planning][:options][:filter_value] = nil
-      if params[:filter_typ] == "all"
-        @sb[:planning][:vms] = {}
-        find_filtered(Vm).sort_by { |v| v.name.downcase }.each { |e| @sb[:planning][:vms][e.id.to_s] = e.name }
-      end
-    end
-    if params[:filter_value]
-      @sb[:planning][:vms] = nil
-      @sb[:planning][:options][:chosen_vm] = nil
-      if params[:filter_value] == "<Choose>"
-        @sb[:planning][:options][:filter_value] = nil
+    vms = nil
+    filter_value = params[:filter_value]
+    if filter_value && filter_value != "<Choose>"
+      case params[:filter_typ]
+      when "host"
+        vms, = Host.find(filter_value).find_filtered_children("vms")
+      when "ems"
+        vms, = ExtManagementSystem.find(filter_value).find_filtered_children("vms")
+      when "cluster"
+        vms, = EmsCluster.find(filter_value).find_filtered_children("all_vms")
+      when "filter"
+        vms = MiqSearch.find(filter_value).results(:userid => current_userid)
       else
-        @sb[:planning][:options][:filter_value] = params[:filter_value]
-        @sb[:planning][:vms] = {}
-        if @sb[:planning][:options][:filter_value]
-          case @sb[:planning][:options][:filter_typ]
-          when "host"
-            vms, count = Host.find(@sb[:planning][:options][:filter_value]).find_filtered_children("vms")
-            vms.each { |v| @sb[:planning][:vms][v.id.to_s] = v.name }
-          when "ems"
-            vms, count = ExtManagementSystem.find(@sb[:planning][:options][:filter_value]).find_filtered_children("vms")
-            vms.each { |v| @sb[:planning][:vms][v.id.to_s] = v.name }
-          when "cluster"
-            vms, count = EmsCluster.find(@sb[:planning][:options][:filter_value]).find_filtered_children("all_vms")
-            vms.each { |v| @sb[:planning][:vms][v.id.to_s] = v.name }
-          when "filter"
-            vms = MiqSearch.find(@sb[:planning][:options][:filter_value]).results(:userid => current_userid)
-            vms.each { |v| @sb[:planning][:vms][v.id.to_s] = v.name }   # Add the VMs to the pulldown hash
-          end
-        end
+        vms = []
       end
+    elsif params[:filter_typ] == "all"
+      vms = find_filtered(Vm).sort_by { |v| v.name.downcase }
     end
-    @sb[:planning][:options][:chosen_vm] = params[:chosen_vm] == "<Choose>" ? nil : params[:chosen_vm] if params[:chosen_vm]
+    @sb[:planning][:vms] = vms.each_with_object({}) { |v, h| h[v.id.to_s] = v.name }
```

Comment:
won't this blow up when `vms` remains `nil` (as initialized at top of method) and not altered because it does not enter `if` or `elsif` statements?